cloudformation-template.yaml

```diff
@@ -1420,19 +1420,58 @@ Resources:
         - Ref: PrivateSubnet2
         - Ref: PrivateSubnet3
   
-  ElastiCacheNode:
-    Type: AWS::ElastiCache::CacheCluster
+  ElastiCacheKmsKey:
+    Type: AWS::KMS::Key
+    Properties:
+      EnableKeyRotation: true
+      KeyPolicy:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action: kms:*
+            Resource: "*"
+            Principal:
+              AWS:
+                - Fn::Sub: arn:aws:iam::${AWS::AccountId}:root
+          - Effect: Allow
```

Comment:
This is not needed for elasticache KMS key.

This was only needed for S3 rest encryption because...it's "different" :)

You can safely remove the second statement

Reply:
Got it. Nevermind :D